packages/datadog-plugin-aws-sdk/src/util.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+const crypto = require('crypto')
+
+/**
+ * Generates a unique hash from an array of strings by joining them with | before hashing.
+ * Used to uniquely identify AWS requests for span pointers.
+ * @param {string[]} components - Array of strings to hash
+ * @returns {string} A 32-character hash uniquely identifying the components
+ */
+function generatePointerHash (components) {
+  // If passing S3's ETag as a component, make sure any quotes have already been removed!
```

Comment:
do we have a test case that checks his for our code?

Reply:
Are you referring to the ETag quotes? I'm not sure if there's an easy way to force that to happen, since `components` is just an array of Strings

Reply:
But we do test the function in `util.spec.js`

Reply:
do we have tests for the code that handles etags they correctly strip the double quote bits?

Reply:
It's not so easy to do that without refactoring the S3 code.

I could open another PR to refactor the `s3.js` file and get the bucket/key/etag extracting in a separate function, and then unit test that function.